### PR TITLE
Styling

### PR DESCRIPTION
**Pull request by FiF0o**
To avoid css bootstrap style leaking through the UI, styled-components library has been added.
Added styling for base components using styled-components for isolation.
Styles applied for the missing components such as buttons, form-fields (text-area) and typography elements (h tags).

### CODE REVIEW

File: config/db.sample.js [hunk 1]
```diff
@@ -1,6 +0,0 @@
-const config = {
```
Comment (panosvoudouris) [inline config/db.sample.js]: Why was this file removed? We need this is as a guide for people who want to use this first time.

File: src/components/button/secondaryButton.jsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  background: black;
+  color: white;
+  &:hover {
+    background: white;
+    color: black;
+    cursor: pointer;
+    }
+  `;
+
+const SecondaryButton = SecondaryWrapper(PrimaryButton);
```
Comment (panosvoudouris) [inline src/components/button/secondaryButton.jsx]: Does this not now create dependency on the PrimaryButton for anything that happens here? Changing a style in the PrimaryButton can break this one.

File: src/components/button/primaryButton.jsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
+/**
+ * Created by jonlazarini on 14/02/17.
+ */
+import React from 'react';
+import Button from './index.jsx';
+import styled from 'styled-components';
+
+
+// Style wrapper for default buttons - Composition
+const PrimaryWrapper = (Comp) => styled(Comp)`
```
Comment (panosvoudouris) [inline src/components/button/primaryButton.jsx]: (Comp) fails the linter. Overall make sure the linter passes before submitting a pull request. Also keep the naming conventions when naming vars e.g. (comp) vs (Comp).

File: src/components/comment/index.jsx [hunk 4]
```diff
@@ -23,17 +26,17 @@ const Comment = ({
...
         { !!currentUserIsOwner &&
-          <button id={commentId} onClick={onUserCommentEdit}>
+          <SecondaryButton id={commentId} onClick={onUserCommentEdit}>
             Edit
-          </button>
+          </SecondaryButton>
         }
         { !!currentUserIsOwner &&
-        <button
+        <HyperLink
```
Comment (panosvoudouris) [inline src/components/comment/index.jsx:33]: When hovering over the Remove button it goes bold and subtly moves the Edit button next to it.

**Review by panosvoudouris — CHANGES_REQUESTED** [review]
Please see individual comments on some changes needed. Overall I'm not so sure about the use of styled.components. Looks like a lot of extra code and bloat vs plain CSS? Opinions @james-nash @jon-ec ?

File: config/db.sample.js (hunk 1 above)
Comment (FiF0o) [reply config/db.sample.js]: Mishandling from my end.

File: src/components/button/secondaryButton.jsx (hunk 2 above)
Comment (FiF0o) [reply src/components/button/secondaryButton.jsx]: You can either modify the style on the primary button which should be named base as it was the intent rather than primary and compose new style(s) from it by extending or overriding the style if necessary.
We could pass the Button instead and was declared this for a concern of not repeating myself for the button base styles.
Renaming the button `base` would make more sense perhaps?

File: src/components/button/primaryButton.jsx (hunk 3 above)
Comment (FiF0o) [reply src/components/button/primaryButton.jsx]: just noticed the linting was not running in the background and there are a few more other errors which I will fix!

File: package.json [hunk 5]
```diff
@@ -72,7 +78,8 @@
     "react": "^15.4.2",
     "react-addons-css-transition-group": "^15.4.2",
     "react-alert": "^1.0.14",
-    "react-dom": "^15.4.2"
+    "react-dom": "^15.4.2",
+    "styled-components": "1.4.3"
```
Comment (crashuniverse) [inline package.json:82]: To maintain consistency, should we keep carets in version. `"styled-components": "^1.4.3"`. And if we need to freeze for a reason in future, we freeze them all.

File: package.json [hunk 6]
```diff
@@ -58,6 +63,7 @@
     "eslint-plugin-jsx-a11y": "^3.0.2",
     "eslint-plugin-react": "^6.9.0",
     "file-loader": "^0.9.0",
+    "ghooks": "2.0.0",
```
Comment (crashuniverse) [inline package.json:66]: Same as below, retaining carets etc.

**Comment by panosvoudouris** [comment]
It would be best to *not* use any packages like styled-components etc but just plain CSS.
